### PR TITLE
Added close all + undo setting + hotkey

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
- Globally increases the shortest duration of snackbar notifications from 2s -> 3s
- Added a "Close All" button in the menu (hotkey is Alt + W + Shift)
- Added a snackbar notification for opening a folder/file you don't have access to
(PR here: https://github.com/zowe/zlux-file-explorer/pull/75)
- Added an "Undo" option to the Close All feature to re-open tabs & files upon accidental missclick

image

image

Signed-off-by: Leanid Astrakou <lastrakou@rocketsoftware.com>

### CODE REVIEW

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts [hunk 1]
```diff
@@ -125,16 +125,16 @@ export class MonacoService {
...
           } else if (err.status === 404) {
-            this.snackBar.open(`${fileNode.name} could not be found`,
-              'Close', { duration: MessageDuration.Short, panelClass: 'center' });
+            this.snackBar.open(`${fileNode.name} could not be found.`,
+              'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
           } else {
-            this.snackBar.open(`${fileNode.name} could not be opened`,
-              'Close', { duration: MessageDuration.Short, panelClass: 'center' });
+            this.snackBar.open(`${fileNode.name} could not be opened.`,
+              'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
```
Comment (DivergentEuropeans) [inline webClient/src/app/editor/code-editor/monaco/monaco.service.ts:137]: Problems and non-OK things are important and deserve time

File: webClient/src/app/shared/message-duration.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@
 */
 
 export class MessageDuration {
-  static Short: number = 2000;
+  static Short: number = 3000;
```
Comment (DivergentEuropeans) [inline webClient/src/app/shared/message-duration.ts:12]: Accessibility

File: webClient/src/app/shared/editor-control/editor-control.service.ts [hunk 3]
```diff
@@ -229,6 +237,32 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
...
+  public undoCloseAllHandler() {
+    this.log.debug("Attempting to restore session with data: ", this.previousSessionData);
+    if (this.previousSessionData.stateCache) {
+      stateCache = this.previousSessionData.stateCache;
+    }
+    if (this.previousSessionData._openFileList) {
+      if (this.previousSessionData._openFileList.length > 0) {
+        this._openFileList.next(this.previousSessionData._openFileList);
+      } else {
+        this._openFileList.next(this.previousSessionData._openFileList);
```
Comment (1000TurquoisePogs) [inline webClient/src/app/shared/editor-control/editor-control.service.ts:261]: these two cases are the same?

**Comment by NakulManchanda** [comment]
@DivergentEuropeans 
One comment is
image

Can this `undo snackbar` come at **bottom of editor app**, rather than outside app bottom of desktop?

File: webClient/src/app/shared/message-duration.ts (hunk 2 above)
Comment (NakulManchanda) [reply webClient/src/app/shared/message-duration.ts:12]: length is good now, but if multiple generated together first one disappear rather quickly,
but good enhancement

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts (hunk 1 above)
Comment (NakulManchanda) [reply webClient/src/app/editor/code-editor/monaco/monaco.service.ts:137]: open two datasets back to back quickly, first one displaced the other immediately.
Good enhancement though, thanks

**Comment by NakulManchanda** [comment]
tried Close all , and undo.
tried both with keyboard shortcut and mouse click
also tried opening files with same name, closing and undo
thanks, working as expected.

**Comment by DivergentEuropeans** [comment]
> @DivergentEuropeans
> One comment is
> image
> 
> Can this `undo snackbar` come at **bottom of editor app**, rather than outside app bottom of desktop?

I actually don't control that, that snackbar comes from Material and that's what it defaults to. We could certainly change it, but we'd have to change it for all snackbars in the Editor & FE since they currently do that. **However**, the added benefit of this is that it's more visible by not hiding within the darkness of the Editor theme 😃 

image

**Pull request merged**